sdk/core/azure-core/src/main/java/com/azure/core/implementation/RestProxy.java

```diff
@@ -154,6 +164,78 @@ public Object invoke(Object proxy, final Method method, Object[] args) {
         }
     }
 
+    private Flux<ByteBuffer> validateLength(final HttpRequest request, final Flux<ByteBuffer> bbFlux) {
+        if (bbFlux == null) {
+            return Flux.empty();
+        }
+        return Flux.defer(new Supplier<Publisher<ByteBuffer>>() {
+            @Override
+            public Publisher<ByteBuffer> get() {
+                Long expectedLength = Long.valueOf(request.headers().value("Content-Length"));
+                List<Integer> bufferLengthList = new ArrayList<>();
```

Comment:
Why are we buffering the running total to a `List` instead of tracking it in a `long`?

Reply:
Lambda only allow us to take in final variable. I have tried Long/long, both not work